spark/dl/src/main/scala/com/intel/analytics/bigdl/models/maskrcnn/MaskRCNN.scala

```diff
@@ -0,0 +1,168 @@
+ */
+
+package com.intel.analytics.bigdl.models.maskrcnn
+
+import com.intel.analytics.bigdl.Module
+import com.intel.analytics.bigdl.models.resnet.{Convolution, Sbn}
+import com.intel.analytics.bigdl.nn._
+import com.intel.analytics.bigdl.nn.abstractnn.Activity
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.{T, Table}
+
+case class MaskRCNNParams(
+  anchorSizes: Array[Float] = Array[Float](32, 64, 128, 256, 512),
+  aspectRatios: Array[Float] = Array[Float](0.5f, 1.0f, 2.0f),
+  anchorStride: Array[Float] = Array[Float](4, 8, 16, 32, 64),
+  preNmsTopNTest: Int = 1000,
+  postNmsTopNTest: Int = 1000,
+  preNmsTopNTrain: Int = 2000,
+  postNmsTopNTrain: Int = 2000,
+  rpnNmsThread: Float = 0.7f,
+  minSize: Int = 0,
+  boxResolution: Int = 7,
+  maskResolution: Int = 14,
+  scales: Array[Float] = Array[Float](0.25f, 0.125f, 0.0625f, 0.03125f),
+  samplingRatio: Int = 2,
+  boxScoreThresh: Float = 0.05f,
+  boxNmsThread: Float = 0.5f,
+  maxPerImage: Int = 100,
+  outputSize: Int = 1024,
+  layers: Array[Int] = Array[Int](256, 256, 256, 256),
+  dilation: Int = 1,
+  useGn: Boolean = false)
+
+class MaskRCNN(val inChannels: Int,
+               val outChannels: Int,
+               val numClasses: Int = 81,
+               val config: MaskRCNNParams = new MaskRCNNParams)(implicit ev: TensorNumeric[Float])
+  extends Container[Activity, Activity, Float] {
+
+    private val ImageInfo : Tensor[Float] = Tensor[Float](2)
+    private val backbone = buildBackbone(inChannels, outChannels)
+    private val rpn = RegionRroposal(inChannels, config.anchorSizes, config.aspectRatios,
+      config.anchorStride, config.preNmsTopNTest, config.postNmsTopNTest, config.preNmsTopNTrain,
+      config.postNmsTopNTrain, config.minSize)
+    private val boxHead = BoxHead(inChannels, config.boxResolution, config.scales,
+      config.samplingRatio, config.boxScoreThresh, config.boxNmsThread, config.maxPerImage,
+      config.outputSize, numClasses)
+    private val maskHead = MaskHead(inChannels, config.maskResolution, config.scales,
+      config.samplingRatio, config.layers, config.dilation, numClasses)
+
+    // add layer to modules
+    modules.append(backbone.asInstanceOf[Module[Float]])
+    modules.append(rpn.asInstanceOf[Module[Float]])
+    modules.append(boxHead.asInstanceOf[Module[Float]])
+    modules.append(maskHead.asInstanceOf[Module[Float]])
+
+    private def buildResNet50(): Module[Float] = {
+
+    def shortcut(nInputPlane: Int, nOutputPlane: Int, stride: Int,
+                 useConv: Boolean = false): Module[Float] = {
+      if (useConv) {
+        Sequential()
+          .add(Convolution(nInputPlane, nOutputPlane, 1, 1, stride, stride))
+          .add(Sbn(nOutputPlane))
+      } else {
+        Identity()
+      }
+    }
+
+    def bottleneck(nInputPlane: Int, internalPlane: Int, nOutputPlane: Int,
```

Comment:
private

Reply:
it's already defined in private `buildResNet50`